gradio/chat_interface.py

```diff
@@ -579,14 +579,12 @@ def response_as_dict(self, response: MessageDict | Message | str) -> MessageDict
             new_response = response
         return cast(MessageDict, new_response)
 
-    async def _submit_fn(
+    def _format_input(
```

Comment:
nit: perhaps we should this method a more descriptive name? it took me a while to understand what its used for. Maybe `process_message_and_trim_history`?